### PR TITLE
Migrate mssql domain service to new module system

### PR DESCRIPTION
**Pull request by DanielFran**
Fix https://github.com/jhipster/jhipster-lite/issues/2119

### CODE REVIEW

**Comment by DanielFran** [comment]
@pascalgrimaud Any clue why CI is failing?

File: tests-ci/config/mssqlapp.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-  "folder": "/tmp/jhlite/mssqlapp",
+  "folder": "/tmp/jhlite/app",
```
Comment (pascalgrimaud) [inline tests-ci/config/mssqlapp.json:2]: Plz rollback this change

**Comment by DanielFran** [comment]
@pascalgrimaud Locally (in windows) I have mssqlapp that build fine, no issue downloading the docker image!?!
Any clue?

**Comment by DanielFran** [comment]
@pascalgrimaud Found the problem, current configuration only works for 2017 mssql docker image version.
This will need to be fine-tuned, but we can work on this after module migration

**Comment by pascalgrimaud** [comment]
Don't understand what is the problem in this migration

Because I used this feature for my current project for my customer, using same version here https://github.com/jhipster/jhipster-lite/blob/main/src/test/resources/generator/dependencies/Dockerfile#L9

And it works well

**Comment by DanielFran** [comment]
@pascalgrimaud I mean the current configuration for testcontainer.
The previous version use mssql v.2019 for prod and v.2017 for integration test with testcontainer (version "latest")

**Comment by pascalgrimaud** [comment]
Oh I see, I'm fine to improve this after migration

**Comment by DanielFran** [comment]
@pascalgrimaud Migration is now done

File: src/main/java/tech/jhipster/lite/generator/server/springboot/database/mssql/domain/MsSQLModuleFactory.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      .set(
+        propertyKey("spring.datasource.url"),
+        propertyValue(
+          "jdbc:tc:sqlserver:///;database=" + properties.projectBaseName().name() + ";trustServerCertificate=true?TC_TMPFS=/testtmpfs:rw"
+        )
+      )
+      .set(propertyKey("spring.datasource.username"), propertyValue("SA"))
+      .set(propertyKey("spring.datasource.password"), propertyValue("yourStrong(!)Password"))
+      .and()
+      .mandatoryReplacements()
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/database/mssql/domain/MsSQLModuleFactory.java:67]: After https://github.com/jhipster/jhipster-lite/pull/2613/files we can updated that to `.integrationTestExtension("MsSQLTestContainerExtension")` so we'll only have to handle conditional replacement (after https://github.com/jhipster/jhipster-lite/issues/2591) once (but need merge for the mongo module before all that so, depending on merge order we'll have to create an issue for that)

File: src/main/java/tech/jhipster/lite/generator/server/springboot/database/mssql/domain/MsSQLModuleFactory.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      .in("src/test/java/" + properties.basePackage().path() + "/IntegrationTest.java")
+      .add(
+        lineBeforeText("import org.springframework.boot.test.context.SpringBootTest;"),
+        "import org.junit.jupiter.api.extension.ExtendWith;"
+      )
+      .add(lineBeforeText("public @interface"), "@ExtendWith(MsSQLTestContainerExtension.class)")
+      .and()
+      .and()
+      .optionalReplacements()
```
Comment (DamnClin) [inline src/main/java/tech/jhipster/lite/generator/server/springboot/database/mssql/domain/MsSQLModuleFactory.java:76]: Same, after https://github.com/jhipster/jhipster-lite/pull/2613/files some improvements to do here

File: src/main/java/tech/jhipster/lite/generator/server/springboot/database/mssql/domain/MsSQLModuleFactory.java (hunk 2 above)
Comment (DanielFran) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/database/mssql/domain/MsSQLModuleFactory.java:67]: Will make changes when working on the upgraded version

File: src/main/java/tech/jhipster/lite/generator/server/springboot/database/mssql/domain/MsSQLModuleFactory.java (hunk 3 above)
Comment (DanielFran) [reply src/main/java/tech/jhipster/lite/generator/server/springboot/database/mssql/domain/MsSQLModuleFactory.java:76]: Idem

**Comment by DamnClin** [comment]
Can I help on this one?

**Comment by pascalgrimaud** [comment]
closing in favor of https://github.com/jhipster/jhipster-lite/pull/2697

**Pull request closed without merging**